### ISSUE TITLE
Use ParamSpec when the wrapper has more keyword-only arguments than wrapped function

### ISSUE THREAD

**Issue by crusaderky** [python/mypy#16479]
[PEP-612](https://peps.python.org/pep-0612/) explains how `Concatenate` can be used to write a wrapper where the output function requires less arguments than the wrapped one.
The opposite, where the output function requires _more_ arguments than the wrapped one, is not clarified by the PEP, nor in my opinion it should, as it seems to be fairly non-controversial.

These work fine with mypy 1.7.0:

```python
from typing import Callable, ParamSpec, TypeVar
P = ParamSpec("P")
R = TypeVar("R")

def w1(f: Callable[P, R], *args: P.args, **kwargs: P.kwargs) -> R:
    return f(*args, **kwargs)

def w2(f: Callable[P, R], extra_arg: int, *args: P.args, **kwargs: P.kwargs) -> R:
    return f(*args, **kwargs)
```

This however doesn't:

```python
def w3(f: Callable[P, R], *args: P.args, extra_arg: int, **kwargs: P.kwargs) -> R:
    return f(*args, **kwargs)
```
```
error: ParamSpec must have "*args" typed as "P.args" and "**kwargs" typed as "P.kwargs"  [valid-type]
```

**Comment by erictraut** [comment]
PEP 612 expressly forbids the signature in `w3` in your example. Here's the exact quote:

> Placing keyword-only parameters between the *args and **kwargs is forbidden.

The "rejected alternatives" section of the PEP gives some indication of why this limitation was added.

Mypy should generate an error for `w3`, but it doesn't currently.

**Comment by ilevkivskyi** [comment]
Duplicate of https://github.com/python/mypy/issues/13966
